modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -892,8 +997,76 @@ func (h *HTTPTransport) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	return nil
 }
 
+// Parse the tokens of a named response matcher.
+//
+//     @name {
+//         header <field> [<value>]
+//         status <code...>
+//     }
+//
+// Or, single line syntax:
+//
+//     @name [header <field> [<value>]] | [status <code...>]
+//
+func (h *Handler) parseNamedResponseMatcher(d *caddyfile.Dispenser, matchers map[string]caddyhttp.ResponseMatcher) error {
```

Comment:
I might be missing something, but `h` is never used here. In fact, it doesn't necessarily have to be a method on `Handler` at all (except maybe for code organization).

```suggestion
func (Handler) parseNamedResponseMatcher(d *caddyfile.Dispenser, matchers map[string]caddyhttp.ResponseMatcher) error {
```

Reply:
Yeah my plan is to move this elsewhere, combined with the code from the precompression PR (which is now merged) which also uses response matchers essentially the same way.

Reply:
Done, moved to a `responsematchers.go`, moved a few things around.